hive/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -0,0 +1,79 @@
+package org.apache.iceberg.hive;
+
+import java.io.IOException;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.iceberg.BaseMetastoreCatalog;
+import org.apache.iceberg.BaseTable;
+import org.apache.iceberg.TableOperations;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.thrift.TException;
+
+import static org.apache.iceberg.BaseMetastoreTableOperations.METADATA_LOCATION_PROP;
+
+public class HiveCatalog extends BaseMetastoreCatalog {
+
+  private final HiveClientPool clients;
+  private final Configuration conf;
+
+  public HiveCatalog(Configuration conf) {
+    super(conf);
+    this.conf = conf;
+    this.clients = new HiveClientPool(2, conf);
+  }
+
+  @Override
+  public void dropTable(TableIdentifier tableIdentifier) {
+    validateTableIdentifier(tableIdentifier);
+    HiveMetaStoreClient hiveMetaStoreClient = this.clients.newClient();
```

Comment:
Looks like `clients.get()` is a more legitimate method to use here? Alternatively, `clients.run()` can be used here.

Reply:
That's correct. `newClient` is protected so that subclasses can provide an implementation,  not so that other classes in this package can call it. The correct way to do this is to use `run`. That handles reconnections and uses the client pool. `get` is also private.